src/main/java/mate/academy/intro/service/UserServiceImpl.java

```diff
@@ -7,22 +7,25 @@
 import mate.academy.intro.mapper.UserMapper;
 import mate.academy.intro.model.User;
 import mate.academy.intro.repository.UserRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @RequiredArgsConstructor
 @Service
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
     private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throws RegistrationException {
-        if (userRepository.existsByEmail(requestDto.getEmail())) {
-            throw new RegistrationException("such email already exists");
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable to complete registration");
         }
-        User user = userMapper.requestDtoToUser(requestDto);
-        userRepository.save(user);
-        return userMapper.userToUserDto(user);
+        User user = new User();
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+        user.setEmail(requestDto.getEmail());
```

Comment:
find role USER from DB and set to user
```suggestion
        user.setRoles(Set.of(role));
```